pkgs/applications/virtualization/vmware-workstation/default.nix

```diff
@@ -0,0 +1,419 @@
+      dest="$out/lib/vmware-installer/$vmware_installer_version"
+
+      mkdir -p $dest
+
+      cp -r $unpacked/vmis* $dest/
+      cp -r $unpacked/sopython $dest/
+      cp -r $unpacked/python $dest/
+      cp -r $unpacked/cdsHelper $dest/
+      cp -r $unpacked/vmware* $dest/
+      cp -r $unpacked/bin $dest/
+      cp -r $unpacked/lib $dest/
+
+      chmod +x $dest/vmis-launcher
+      chmod +x $dest/sopython/*
+      chmod +x $dest/python/init.sh
+      chmod +x $dest/vmware-*
+
+      ln -s $dest/vmware-installer $out/bin/vmware-installer
+
+      mkdir -p $out/etc/vmware-installer
+
+      cat <<EOF > $out/etc/vmware-installer/bootstrap
+      VMWARE_INSTALLER="$dest"
+      VERSION="$vmware_installer_version" # For backwards compability
+      VMISVERSION="$vmware_installer_version"
+      VMISBUILDNUM="${build}"
+      VMISPYVERSION="39"
+      EOF
+
+      mkdir -p $out/etc/vmware-installer/components
+
+      # create database of vmware guest tools (avoids vmware fetching them later)
+      local database_filename=$out/etc/vmware-installer/database
+      echo -n "" > $database_filename
+      sqlite3 "$database_filename" "CREATE TABLE settings(key VARCHAR PRIMARY KEY, value VARCHAR NOT NULL, component_name VARCHAR NOT NULL);"
+      sqlite3 "$database_filename" "INSERT INTO settings(key,value,component_name) VALUES('db.schemaVersion','2','vmware-installer');"
+      sqlite3 "$database_filename" "CREATE TABLE components(id INTEGER PRIMARY KEY, name VARCHAR NOT NULL, version VARCHAR NOT NULL, buildNumber INTEGER NOT NULL, component_core_id INTEGER NOT NULL, longName VARCHAR NOT NULL, description VARCHAR, type INTEGER NOT NULL);"
+      for folder in unpacked/**/.installer ; do
+        component="$(basename $(dirname $folder))"
+        local component_version=$(cat unpacked/$component/manifest.xml | grep -oPm1 "(?<=<version>)[^<]+")
+        local component_core_id=$([ "$component" == "vmware-installer" ] && echo "-1" || echo "1")
+        local type=$([ "$component" == "vmware-workstation" ] && echo "0" || echo "1")
+        sqlite3 "$database_filename" "INSERT INTO components(name,version,buildNumber,component_core_id,longName,description,type) VALUES(\"$component\",\"$component_version\",\"${build}\",$component_core_id,\"$component\",\"$component\",$type);"
+        mkdir -p $out/etc/vmware-installer/components/$component
+        cp -r $folder/* $out/etc/vmware-installer/components/$component
+      done
+
+    ''}
+
+
+
+    ## VMware Bootstrap
+    echo "Installing VMware Bootstrap"
+    cat <<EOF > $out/etc/vmware/bootstrap
+    PREFIX="$out"
+    BINDIR="$out/bin"
+    SBINDIR="$out/sbin"
+    LIBDIR="$out/lib"
+    DATADIR="$out/share"
+    SYSCONFDIR="$out/etc"
+    DOCDIR="$out/share/doc"
+    MANDIR="$out/share/man"
+    INCLUDEDIR="$out/include"
+    INITDIR="$out/etc"
+    INITSCRIPTDIR="$out/etc/init.d"
+    EOF
+
+
+
+    ## VMware Config
+    echo "Installing VMware Config"
+
+    cat <<EOF > $out/etc/vmware/config
```

Comment:
I don't like that we are piping to so many files. It would probably be easier to create them with a patch and substitute the variables.

Reply:
Piping is now removed, 3 files: `vmware-installer-boostrap`,`vmware-boostrap`,`vmware-config` now used instead with sed to substite variables in them.